plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/glue/converter/GlueStatConverter.java

```diff
@@ -0,0 +1,291 @@
+                break;
+            case DATE:
+                DateColumnStatisticsData catalogDateData = new DateColumnStatisticsData();
+                statistics.getDateStatistics().ifPresent(dateStatistics -> {
+                    dateStatistics.getMin().ifPresent(value -> catalogDateData.setMinimumValue(localDateToDate(value)));
+                    dateStatistics.getMax().ifPresent(value -> catalogDateData.setMaximumValue(localDateToDate(value)));
+                });
+                statistics.getNullsCount().ifPresent(catalogDateData::setNumberOfNulls);
+                toMetastoreDistinctValuesCount(statistics.getDistinctValuesCount(), statistics.getNullsCount()).ifPresent(catalogDateData::setNumberOfDistinctValues);
+                catalogColumnStatisticsData.setType(ColumnStatisticsType.DATE.toString());
+                catalogColumnStatisticsData.setDateColumnStatisticsData(catalogDateData);
+                break;
+            case DECIMAL:
+                DecimalColumnStatisticsData catalogDecimalData = new DecimalColumnStatisticsData();
+                statistics.getDecimalStatistics().ifPresent(decimalStatistics -> {
+                    decimalStatistics.getMin().ifPresent(value -> catalogDecimalData.setMinimumValue(bigDecimalToGlueDecimal(value)));
+                    decimalStatistics.getMax().ifPresent(value -> catalogDecimalData.setMaximumValue(bigDecimalToGlueDecimal(value)));
+                });
+                statistics.getNullsCount().ifPresent(catalogDecimalData::setNumberOfNulls);
+                toMetastoreDistinctValuesCount(statistics.getDistinctValuesCount(), statistics.getNullsCount()).ifPresent(catalogDecimalData::setNumberOfDistinctValues);
+                catalogColumnStatisticsData.setType(ColumnStatisticsType.DECIMAL.toString());
+                catalogColumnStatisticsData.setDecimalColumnStatisticsData(catalogDecimalData);
+                break;
+            case FLOAT:
+            case DOUBLE:
+                DoubleColumnStatisticsData catalogDoubleData = new DoubleColumnStatisticsData();
+                statistics.getDoubleStatistics().ifPresent(doubleStatistics -> {
+                    doubleStatistics.getMin().ifPresent(catalogDoubleData::setMinimumValue);
+                    doubleStatistics.getMax().ifPresent(catalogDoubleData::setMaximumValue);
+                });
+                statistics.getNullsCount().ifPresent(catalogDoubleData::setNumberOfNulls);
+                toMetastoreDistinctValuesCount(statistics.getDistinctValuesCount(), statistics.getNullsCount()).ifPresent(catalogDoubleData::setNumberOfDistinctValues);
+                catalogColumnStatisticsData.setType(ColumnStatisticsType.DOUBLE.toString());
+                catalogColumnStatisticsData.setDoubleColumnStatisticsData(catalogDoubleData);
+                break;
+            case BYTE:
+            case SHORT:
+            case INT:
+            case LONG:
+            case TIMESTAMP:
```

Comment:
The intent here is to handle all the types (`SHORT,INT,LONG,TIMESTAMP`) Using the same logic by leveraging java `switch` multiple cases, so in this case all the defined types are written with the `LongColumnStatisticsData` object to glue, Glue doesn't support short, int and timestamp values. 

Should we make this more explicit ? 
Another possible change is to handle `timestamp` type as `Date` instead of `long` but I'm not sure if this can help